components/script/dom/htmlmediaelement.rs

```diff
@@ -2069,6 +2069,15 @@ impl HTMLMediaElement {
             }
         }
     }
+
+    pub(crate) fn same_origin(&self, origin: &MutableOrigin) -> bool {
+        // TODO: Step 8.6. Update the media data with the contents of response.
+        // <https://html.spec.whatwg.org/multipage/#concept-media-load-resource>
+        match self.resource_url.borrow().as_ref() {
```

Comment:
This will treat a request that is initially same-origin but redirects to a cross-origin URL as same-origin, which explains why there are tests for that case that are failing after these changes. Can we implement the correct fix here, by storing the Metadata.final_url that's available in process_response and using that to compare instead of self.resource_url?

Reply:
https://github.com/servo/servo/blob/cf3ad2de774c8f8022135b0a9d466c197d961aa2/components/script/stylesheet_loader.rs#L132-L138 is another way to handle this situation.

Reply:
Done